### PR TITLE
Move PasswordHistoryModel into api-integration from openedX.

### PR DESCRIPTION
**Pull request by attiyaIshaque**
### Description:
Password History Model and its tests moved to api-integration from openedX.

### CODE REVIEW

File: edx_solutions_api_integration/models.py [hunk 1]
```diff
@@ -3,6 +3,9 @@
 """ Database ORM models managed by this Django app """
 from django.contrib.auth.models import Group, User
 from django.db import models
+from django.utils import timezone
+from django.contrib.auth.hashers import make_password
+from django.conf import settings
```
Comment (ihtram) [inline edx_solutions_api_integration/models.py]: Can we organize imports here?

File: edx_solutions_api_integration/models.py [hunk 2]
```diff
@@ -156,3 +159,187 @@ class APIUser(User):
...
+            hash_elements = entry.password.split('$')
+            algorithm = hash_elements[0]
+            if algorithm == 'pbkdf2_sha256':
+                hashed_password = make_password(new_password, hash_elements[2])
+            elif algorithm == 'sha1':
+                hashed_password = make_password(new_password, hash_elements[1])
+            else:
+                # This means we got something unexpected. We don't want to throw an exception, but
+                # log as an error and basically allow any password reuse
+                AUDIT_LOG.error('''
```
Comment (ihtram) [inline edx_solutions_api_integration/models.py]: It seem like import missing for AUDIT_LOG

File: edx_solutions_api_integration/models.py [hunk 3]
```diff
@@ -156,3 +159,187 @@ class APIUser(User):
...
+        self.password = user.password
+        self.save()
+
+    @classmethod
+    def is_student_password_reuse_restricted(cls):
+        """
+        Returns whether the configuration which limits password reuse has been turned on
+        """
+        min_diff_pw = settings.ADVANCED_SECURITY_CONFIG.get(
+            'MIN_DIFFERENT_STUDENT_PASSWORDS_BEFORE_REUSE', 0
```
Comment (ihtram) [inline edx_solutions_api_integration/models.py]: We need to check if these variables are defined in ironwood or not? 
ADVANCED_SECURITY_CONFIG
MIN_DIFFERENT_STUDENT_PASSWORDS_BEFORE_REUSE

Comment (attiyaIshaque) [reply edx_solutions_api_integration/models.py]: These variables are not defined in ironwood we have to add it explicitly.

File: edx_solutions_api_integration/models.py (hunk 2 above)
Comment (attiyaIshaque) [reply edx_solutions_api_integration/models.py]: Okay i will add that.

File: edx_solutions_api_integration/sessions/views.py [hunk 4]
```diff
@@ -22,8 +22,10 @@
 from edx_solutions_api_integration.utils import generate_base_uri
 
 from edx_solutions_api_integration.users.serializers import SimpleUserSerializer
-# TODO: PasswordHistory is removed from openedx, remove it from here as well
-from student.models import LoginFailures  #, PasswordHistory
+from edx_solutions_api_integration.models import PasswordHistory
+from student.models import (
```
Comment (ihtram) [inline edx_solutions_api_integration/sessions/views.py]: it should be a single liner like 
from student.models import LoginFailures

File: edx_solutions_api_integration/models.py [hunk 5]
```diff
@@ -156,3 +162,187 @@ class APIUser(User):
...
+
+class PasswordHistory(models.Model):
+    """
+    This model will keep track of past passwords that a user has used
+    as well as providing contraints (e.g. can't reuse passwords)
+    """
+    class Meta(object):
+        app_label = "student"
+
+    user = models.ForeignKey(User)
```
Comment (shafqatfarhan) [inline edx_solutions_api_integration/models.py]: This should be `user = models.ForeignKey(User, on_delete=models.CASCADE)`. This change should make the deprecation warnings in the current Django version go away.

File: edx_solutions_api_integration/test_password_history.py [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+        # ensure the rows were actually created and stored the passwords
+        self.assertTrue(PasswordHistory.objects.filter(user_id=user.id).exists())
+        for row in PasswordHistory.objects.filter(user_id=user.id):
+            self.assertFalse(row.password == "")
+
+        # retire the user and ensure that the rows are still present, but with no passwords
+        PasswordHistory.retire_user(user.id)
+        self.assertTrue(PasswordHistory.objects.filter(user_id=user.id).exists())
+        for row in PasswordHistory.objects.filter(user_id=user.id):
+            self.assertEqual(row.password, "")
```
Comment (shafqatfarhan) [inline edx_solutions_api_integration/test_password_history.py]: EOF line missing

**Comment by attiyaIshaque** [comment]
@ihtram , @shafqatfarhan I have addressed your comments.

File: edx_solutions_api_integration/models.py (hunk 3 above)
Comment (ihtram) [reply edx_solutions_api_integration/models.py]: @attiyaIshaque where these variables have been added?

Comment (attiyaIshaque) [reply edx_solutions_api_integration/models.py]: @ihtram In lms.env.json.

**Pull request merged**